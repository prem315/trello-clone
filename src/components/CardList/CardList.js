import React, { useState, useRef } from "react";
import Card from "../Card/Card";
import Input from "../Input/input";
import IconButton from "../IconButton/IconButton";
import "./CardList.scss";

export default function CardList({
    card,
    addTaskToCard,
    handleTaskInputChange,
    taskInput,
    cardId,
    key,
    delteTaskToCard,
    editTaskToCard,
    dragAndDropTask
}) {
    
    const [val, setVal] = useState("");
    // const [dragId, setDragId] = useState();
    const [dragCard, setDragCard] = useState("");

    

    const handleDrag = (ev, card) => {
       
        // setDragCard(card)
        ev.dataTransfer.setData("dragCardId",card.id);
        // setDragId(ev.currentTarget.id);
        ev.dataTransfer.setData("dragId", ev.currentTarget.id)
        
    };

    const handleDrop = (ev, card, task) => {
        const dropId = ev.currentTarget.id
        const dragcardId=ev.dataTransfer.getData("dragCardId");
        // console.log(dragId)
        const dragId = ev.dataTransfer.getData("dragId");
        dragAndDropTask(dragId, dropId, card, task, dragcardId)
    };

    const handleChange = (val) => {
        setVal(val);
    };
    const handleAddTaskToCard = (val, card) => {
        addTaskToCard(val, card);
        setVal("");
    };
    const delteTaskFromCard = (card, task) => {
        delteTaskToCard(card, task)
    }
    return (
        <div className="cardList" key={cardId} >
            <div className="cardLable">{card.label}</div>
            <div className="cardContainer">
                {card.tasks.map((task, index) => {
                    return <Card
                        index={index} 
                        task={task} 
                        key={task.id} 
                        delteTaskToCard={() => delteTaskFromCard(card, task)} 
                        editTaskToCard={(e) => editTaskToCard(e, card, task)}
                        //dragAndDropTask={(dragItem, dragOverItem, task) => dragAndDropTask(dragItem, dragOverItem, card)}
                        handleDrag={(e) => handleDrag(e, card)}
                        handleDrop={(e) => handleDrop(e, card, task)}
                    />;
                })}
            </div>
            <div className="inputContainer" key={cardId}>
                <Input
                    value={val}
                    handleTaskInputChange={(val) => handleChange(val)}
                    key={cardId}
                />
                <IconButton
                    icon={"plus"}
                    handleClick={() => handleAddTaskToCard(val, card)}
                    key={cardId}
                />
            </div>
        </div>
    );
}
