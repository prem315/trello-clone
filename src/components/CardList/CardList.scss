.cardList {
    width: 270px;
    padding: 10px;
    height: 100%;
    overflow: hidden;
    min-height: 100px;
    background-color: #e2e4e6;
    box-shadow: 1px 1px 1px rgb(0 0 0 / 50%);
    border-radius: 3px;
    padding-bottom: 20px;
    margin-right: 20px;
    vertical-align: top;

    .cardLable {
        text-align: center;
        font-weight: 700;
        white-space: normal;
        width: 100%;
        padding: 5px;
    }

    .cardContainer {
        padding: 5px;
    }

    .inputContainer {
        padding: 5px;
        display: flex;
        align-items: center;
        justify-content: space-around;
    }
}