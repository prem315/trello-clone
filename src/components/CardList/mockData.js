import { v4 as uuidv4 } from "uuid";

export const mockCardData = {
	id: uuidv4(),
	label: "Todo List",
	tasks: [],
};