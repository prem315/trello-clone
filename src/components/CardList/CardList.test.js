import { render, fireEvent, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { v4 as uuidv4 } from "uuid";
import CardList from "./CardList";
import React from "react";
import { mockCardData } from "./mockData";

describe("testing <CardList /> component", () => {
	const mockSetState = jest.fn();

	jest.mock("react", () => ({
		useState: (initial) => [initial, mockSetState],
	}));

	test("should add task to cardList component", () => {
		const mockAddTask = jest.fn();
		const { debug, rerender } = render(
			<CardList
				delteTaskToCard={jest.fn()}
				card={mockCardData}
				cardId=""
				editTaskToCard={jest.fn()}
				dragAndDropTask={jest.fn()}
				editLableChange={jest.fn()}
				editTask={jest.fn()}
				deleteCardList={jest.fn()}
				addTaskToCard={mockAddTask}
			/>
		);

		const addCardButton = screen.getByTestId("add-card-button");
		const taskInput = screen.getByTestId("add-card-input");
		expect(taskInput).toBeInTheDocument();
		fireEvent.change(taskInput, { target: { value: "addingcard" } });
		fireEvent.click(addCardButton);

		expect(mockAddTask).toBeCalledWith("addingcard", mockCardData);
	});

	test("it should match snapshot", () => {
		const MockCard = render(
			<CardList
				delteTaskToCard={jest.fn()}
				card={mockCardData}
				cardId=""
				editTaskToCard={jest.fn()}
				dragAndDropTask={jest.fn()}
				editLableChange={jest.fn()}
				editTask={jest.fn()}
				deleteCardList={jest.fn()}
				addTaskToCard={jest.fn()}
			/>
		);

		expect(MockCard).toMatchSnapshot();
	});
});