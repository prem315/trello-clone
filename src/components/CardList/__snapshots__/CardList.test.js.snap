// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <CardList /> component it should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cardList"
        data-testid="Todo List"
      >
        <div
          class="cardLable"
        >
          <section
            data-testid="editable-input"
          >
            <div>
              <span>
                Todo List
              </span>
            </div>
          </section>
        </div>
        <div
          class="cardContainer"
        />
        <div
          class="inputContainer"
        >
          <input
            aria-label="cost-input"
            class="inp"
            data-testid="add-card-input"
            placeholder="add task here"
            value=""
          />
          <button
            class="icon-button-img"
            data-testid="add-card-button"
            disabled=""
            icon="plus"
          >
            <img
              alt="plus"
              aria-disabled="true"
              aria-label="cost-icon-button"
              class="image-button"
              src="plus.png"
            />
          </button>
        </div>
        <div
          class="delete-btn-container"
        >
          <button
            class="delete-button"
            data-testid="Todo List-delete-button"
            type="delte"
          >
            Delete
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cardList"
      data-testid="Todo List"
    >
      <div
        class="cardLable"
      >
        <section
          data-testid="editable-input"
        >
          <div>
            <span>
              Todo List
            </span>
          </div>
        </section>
      </div>
      <div
        class="cardContainer"
      />
      <div
        class="inputContainer"
      >
        <input
          aria-label="cost-input"
          class="inp"
          data-testid="add-card-input"
          placeholder="add task here"
          value=""
        />
        <button
          class="icon-button-img"
          data-testid="add-card-button"
          disabled=""
          icon="plus"
        >
          <img
            alt="plus"
            aria-disabled="true"
            aria-label="cost-icon-button"
            class="image-button"
            src="plus.png"
          />
        </button>
      </div>
      <div
        class="delete-btn-container"
      >
        <button
          class="delete-button"
          data-testid="Todo List-delete-button"
          type="delte"
        >
          Delete
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
